apps/storefront/src/components/cart/cart-panels/CartShippingPanel.tsx

```diff
@@ -8,61 +8,74 @@ import {
   Text,
   VStack,
 } from "@chakra-ui/react";
-import { Address, OrderWorksheet } from "ordercloud-javascript-sdk";
-import React from "react";
-import { useShippingMethods } from "../../../hooks/useShippingMethods";
+import { useShopper } from "@ordercloud/react-sdk";
+import {
+  Address,
+  IntegrationEvents,
+  OrderShipMethodSelection,
+  ShipMethod,
+} from "ordercloud-javascript-sdk";
+import React, { useState } from "react";
 
 interface CartShippingPanelProps {
-  orderWorksheet: OrderWorksheet;
   shippingAddress: Address;
   handleNextTab: () => void;
   handlePrevTab: () => void;
 }
 
 const CartShippingPanel: React.FC<CartShippingPanelProps> = ({
-  orderWorksheet,
-  shippingAddress,
   handleNextTab,
 }) => {
-  const {
-    availableMethods,
-    selectedMethodIndex,
-    selectShipMethod,
-    error,
-    loading,
-  } = useShippingMethods({ orderWorksheet, shippingAddress });
+  const { orderWorksheet, refreshWorksheet, calculateOrder } = useShopper();
+  const [shipMethodID, setShipMethodID] = useState<string>();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
 
-  const handleShippingMethod = async () => {
+  const handleSelectShipMethod = async () => {
     const orderID = orderWorksheet?.Order?.ID;
     const shipEstimateID =
       orderWorksheet?.ShipEstimateResponse?.ShipEstimates?.at(0)?.ID;
 
-    if (!orderID || !selectedMethodIndex || !shipEstimateID) {
+    if (!orderID || !shipEstimateID) {
       console.error("Missing required data for ship method selection.");
       return;
     }
 
-    const selectedMethod = availableMethods[parseInt(selectedMethodIndex)];
-
-    if (!selectedMethod) {
+    if (!shipMethodID) {
       console.error("Selected method not found.");
       return;
     }
-  };
 
-  const handleChange = (value: string) => {
-    selectShipMethod(value);
-    const selected = availableMethods[parseInt(value)];
-    console.log("Selected shipping method:", selected);
-    handleNextTab();
+    const shipMethodSelection: OrderShipMethodSelection = {
+      ShipMethodSelections: [
+        { ShipEstimateID: shipEstimateID, ShipMethodID: shipMethodID },
+      ],
+    };
+
+    try {
+      setLoading(true);
+      await IntegrationEvents.SelectShipmethods(
```

Comment:
you can call selectShipMethods(shipMethodSelection) here from the useShopper hook.  also, i'm pretty sure you don't need the refreshWorksheet() call here, since that happens in the onsuccess callback of the calculateorder hook in useShopper. same with the refresh call in the ShoppingCart component